deploy/helm/elastic-agent/templates/integrations/_kubernetes/_kubernetes_logs_containers.tpl

```diff
@@ -16,12 +16,12 @@ Config input for container logs
     namespace: {{ .Values.kubernetes.namespace }}
   use_output: {{ .Values.kubernetes.output }}
   streams:
-  - id: kubernetes-container-logs-${kubernetes.pod.name}-${kubernetes.container.id}
+  - id: kubernetes-container-logs-${kubernetes.namespace}-${kubernetes.pod.name}-${kubernetes.container.name}
     data_stream:
       dataset: kubernetes.container_logs
       type: logs
```

Comment:
As long as this translates to a running `filestream` input and we don't switch from 8.x to 9.x the change in the PR should cause no re-indexing.

Starting with 9.0 we will have fingerprint as a default file identity which is based on the file content, not its filesystem metadata.

Reply:
> Does this mean that an upgrade from 8.x to 9.x will result in a re-index of all files? That would be very bad.

No, we also implemented a feature to migrate the state from the old default (`native` that uses inode + device ID) to the new default `fingerprint`. If you're curious, [here is the PR](https://github.com/elastic/beats/pull/41762).